### PR TITLE
Add ability to delete/destroy tweets to V1 API

### PR DESCRIPTION
**Pull request by HankG**
- Adds ability to delete a tweet based on the Tweet's Long ID
- Updates the Update test to be an Update *and* Delete so that the timeline has proper clean up and tests both functions

### CODE REVIEW

File: v1/v1-api/src/commonMain/kotlin/com/codingfeline/twitter4kt/v1/api/statuses/destroy.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * This is rate limited by Twitter.
+ *
+ * [Twitter API reference](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/post-and-engage/api-reference/post-statuses-destroy-id)
+ *
+ * @param id The ID of the tweet authenticated user's tweet that should be deleted
+ */
+@OptIn(Twitter4ktInternalAPI::class)
+public suspend fun StatusesApi.destroy(
+    id: Long
```
Comment (yshrsmz) [inline v1/v1-api/src/commonMain/kotlin/com/codingfeline/twitter4kt/v1/api/statuses/destroy.kt:40]:
```suggestion
    id: String
```

We should prefer `id_str` over `id`, so this needs to be String.
see: https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/overview/tweet-object

Also, could you add `trimUser` parameter for completeness?

**Review by yshrsmz — CHANGES_REQUESTED** [review]

File: v1/v1-api/src/commonMain/kotlin/com/codingfeline/twitter4kt/v1/api/statuses/destroy.kt (hunk 1 above)
Comment (HankG) [reply v1/v1-api/src/commonMain/kotlin/com/codingfeline/twitter4kt/v1/api/statuses/destroy.kt:40]: Trim user breaks the Kotlinx serialization right now. It complains about standard fields "name" being the first. I'm guessing that the field "user" is returned but not set which causes a cascade of creating a null user. I can add that back but we'll have to make nullable every field.

Comment (HankG) [reply v1/v1-api/src/commonMain/kotlin/com/codingfeline/twitter4kt/v1/api/statuses/destroy.kt]: I've updated the User object to work under trimUser and added the argument to the method.

File: v1/v1-model/src/commonMain/kotlin/com/codingfeline/twitter4kt/v1/model/user/User.kt [hunk 2]
```diff
@@ -27,24 +27,24 @@ import kotlinx.serialization.Serializable
 public data class User(
     @SerialName("id") val id: Long,
     @SerialName("id_str") val idStr: String,
-    @SerialName("name") val name: String,
-    @SerialName("screen_name") val screenName: String,
-    @SerialName("location") val location: String?,
+    @SerialName("name") val name: String? = null,
```
Comment (yshrsmz) [inline v1/v1-model/src/commonMain/kotlin/com/codingfeline/twitter4kt/v1/model/user/User.kt:30]: Oh, now I see what you mean.
This indeed does not looks good 🤔

Comment (HankG) [reply v1/v1-model/src/commonMain/kotlin/com/codingfeline/twitter4kt/v1/model/user/User.kt:30]: The options are to not support trimUser, do the null stuff above, or have them fill in default empty fields. I'm currently doing the nullification.

**Review by yshrsmz — APPROVED** [review]
Ok, I made the decision.
Let's keep it simple for now :)
Ignore `trim_user` and think about it later.

I'm going to revert your last commit and merge this PR.

**Comment by yshrsmz** [comment]
Thanks!

**Pull request merged**